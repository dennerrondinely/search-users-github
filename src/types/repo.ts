export interface IOwner {
  avatar_url: string;
}

export interface IRepo {
  name: string;
  description: string | null;
  html_url: string;
  stargazers_count: number;
  owner: IOwner;
}

export type Repo = {
  name: string;
  description: string;
  url: string;
  stargazers: number;
  avatarUrl: string;
};
