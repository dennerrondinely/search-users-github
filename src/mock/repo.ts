import { Repo } from 'types/repo';

export const repo: Repo = {
  name: 'awesome-svg',
  description: 'A curated list of SVG.',
  stargazers: 4205,
  url: 'https://github.com/willianjusten/awesome-svg',
  avatarUrl: 'https://avatars.githubusercontent.com/u/3991845?v=4'
};

export const repos: Repo[] = [
  {
    name: 'renoise',
    description: 'ReNoise Dracula Theme',
    stargazers: 12,
    url: 'https://github.com/dracula/renoise',
    avatarUrl: 'https://avatars.githubusercontent.com/u/19436447?v=4'
  },
  {
    name: 'awesome-svg',
    description: 'A curated list of SVG.',
    stargazers: 4205,
    url: 'https://github.com/willianjusten/awesome-svg',
    avatarUrl: 'https://avatars.githubusercontent.com/u/3991845?v=4'
  },
  {
    name: 'bootcamp-rocketseat',
    description: '_',
    stargazers: 0,
    url: 'https://github.com/dracula/renoise',
    avatarUrl: 'https://avatars.githubusercontent.com/u/14242874?v=4'
  }
];
