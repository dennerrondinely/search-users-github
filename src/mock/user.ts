import { User } from 'types/user';

const user: User = {
  name: 'DÃªnner Rondinely',
  login: 'dennerrondinely',
  bio: 'Geek and JavaScript enthusiast ğŸ•¶',
  avatarUrl: 'https://avatars.githubusercontent.com/u/14242874?v=4',
  reposNumber: 39
};

export default user;
