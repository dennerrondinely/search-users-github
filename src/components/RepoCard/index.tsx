import Icon from 'components/Icon';
import React from 'react';
import { Repo } from 'types/repo';

import * as S from './styles';

export type RepoCardProps = {
  repo: Repo;
};

const RepoCard = ({ repo }: RepoCardProps) => {
  return (
    <S.Wrapper>
      <S.AvatarWrapper>
        <S.Avatar src={repo.avatarUrl} alt={repo.name} />
      </S.AvatarWrapper>
      <S.ContentWrapper>
        <S.Title title={repo.name}>{repo.name}</S.Title>
        <S.Description title={repo.description}>
          {repo.description}
        </S.Description>
        <S.InfoWrapper>
          <S.LinkButton href={repo.url} target="_blank">
            Ver no github
          </S.LinkButton>
          <S.Stargazers title="NÃºmero de estrelas">
            {repo.stargazers}
            <Icon icon="star" color="yellow" size={14} />
          </S.Stargazers>
        </S.InfoWrapper>
      </S.ContentWrapper>
    </S.Wrapper>
  );
};

export default RepoCard;
