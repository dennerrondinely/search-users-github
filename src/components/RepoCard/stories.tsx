import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import RepoCard, { RepoCardProps } from '.';
import { repo } from 'mock/repo';

export default {
  title: 'RepoCard',
  component: RepoCard,
  argTypes: {
    onRepoClick: { action: 'clicked' },
    repo: { type: '' }
  }
} as Meta;

export const Default: Story<RepoCardProps> = (args) => <RepoCard {...args} />;

Default.args = {
  repo
};
