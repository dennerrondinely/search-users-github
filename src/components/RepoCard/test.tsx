import React from 'react';
import { render, screen } from 'utils/tests/helpers';

import { repo } from 'mock/repo';

import RepoCard from '.';

describe('<RepoCard />', () => {
  it('should render a card repo with his information', () => {
    const { container } = render(<RepoCard repo={repo} />);
    expect(screen.getByRole('img', { name: repo.name })).toHaveAttribute(
      'src',
      repo.avatarUrl
    );

    expect(
      screen.getByRole('heading', { name: repo.name })
    ).toBeInTheDocument();

    expect(screen.getByText(repo.description)).toBeInTheDocument();

    expect(screen.getByText(repo.stargazers)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
