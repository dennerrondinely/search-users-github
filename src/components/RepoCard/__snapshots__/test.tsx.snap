// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoCard /> should render a card repo with his information 1`] = `
.c9 {
  height: 14px;
  width: 14px;
  height: 14px;
  width: 14px;
  stroke: #f9d671;
  fill: #f9d671;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2rem;
  position: relative;
  width: 100%;
  max-width: 34rem;
}

.c1 {
  border-radius: 1rem;
  overflow: hidden;
  height: 9rem;
  width: 9rem;
}

.c2 {
  height: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  gap: 1rem;
  width: calc(100% - 11rem);
}

.c4 {
  white-space: nowrap;
  width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 1.8rem;
  color: #241911;
}

.c5 {
  white-space: nowrap;
  width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 1.2rem;
  color: #7c7570;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7 {
  border: none;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #2B716E;
  background-color: #0000;
  font-size: 1.2rem;
}

.c8 {
  min-width: 3rem;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  line-height: 1.4rem;
  border-radius: 1.4rem;
  color: #2B716E;
  font-size: 1.2rem;
  border: 0.1rem dashed #e8efef;
  padding: 0.4rem 0.8rem;
  gap: 0.4rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="awesome-svg"
      class="c2"
      src="https://avatars.githubusercontent.com/u/3991845?v=4"
    />
  </div>
  <div
    class="c3"
  >
    <h2
      class="c4"
      title="awesome-svg"
    >
      awesome-svg
    </h2>
    <span
      class="c5"
      title="A curated list of SVG."
    >
      A curated list of SVG.
    </span>
    <div
      class="c6"
    >
      <a
        class="c7"
        href="https://github.com/willianjusten/awesome-svg"
        target="_blank"
      >
        Ver no github
      </a>
      <span
        class="c8"
        title="repo stars"
      >
        4205
        <svg
          class="c9"
          color="yellow"
          size="14"
          viewBox="0 0 16 15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.29915 11.9694L3.75063 14.3612L4.61937 9.29614L0.93811 5.70846L6.02378 4.96941L8.29915 0.361206L10.5738 4.96941L15.6595 5.70846L11.9789 9.29614L12.8484 14.3612L8.29915 11.9694Z"
            stroke-width="0"
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
